From d15d9fd02575ecfada92d42f655940c4f10af842 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Wed, 20 Feb 2019 07:52:31 +0000
Subject: drm: bridge: dw-hdmi: Fix overflow workaround for Rockchip SoCs
Git-commit: d15d9fd02575ecfada92d42f655940c4f10af842
Patch-mainline: v5.1-rc7
References: bsc#1113722

The Rockchip RK3288 SoC (v2.00a) and RK3328/RK3399 SoCs (v2.11a) have
also been identified as needing this workaround with a single iteration.

Fixes: be41fc55f1aa ("drm: bridge: dw-hdmi: Handle overflow workaround based on device version")
Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
Tested-by: Heiko Stueber <heiko@sntech.de>
Signed-off-by: Andrzej Hajda <a.hajda@samsung.com>
Link: https://patchwork.freedesktop.org/patch/msgid/AM3PR03MB0966818FAAAE6192FF4ED11AAC7D0@AM3PR03MB0966.eurprd03.prod.outlook.com
Acked-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -1597,6 +1597,8 @@ static void dw_hdmi_clear_overflow(struc
 	 * 4 and 1 iterations respectively.
 	 * The Amlogic Meson GX SoCs (v2.01a) have been identified as needing
 	 * the workaround with a single iteration.
+	 * The Rockchip RK3288 SoC (v2.00a) and RK3328/RK3399 SoCs (v2.11a) have
+	 * been identified as needing the workaround with a single iteration.
 	 */

 	switch (hdmi->version) {
@@ -1604,7 +1606,9 @@ static void dw_hdmi_clear_overflow(struc
 		count = 4;
 		break;
 	case 0x131a:
+	case 0x200a:
 	case 0x201a:
+	case 0x211a:
 		count = 1;
 		break;
 	default:
